File: No_1493/Longest Subarray of 1's After Deleting One Element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int len = nums.size();
        vector<int> leftOnes(len), rightOnes(len);
        
        leftOnes[0] = nums[0];
        for(int i=1; i<len; i++) {
            if(nums[i] == 1) {
                leftOnes[i] = leftOnes[i-1] + 1;
            }
        }
        
        rightOnes[len-1] = nums[len-1];
        for(int i=len-2; i>=0; i--) {
            if(nums[i] == 1) {
                rightOnes[i] = rightOnes[i+1] + 1;
            }
        }
        
        int ret = 0;
        for(int i=0; i<len; i++) {
            if(nums[i] == 0) {
                int tmp = 0;
                if(i != 0) tmp += leftOnes[i-1];
                if(i != len-1) tmp += rightOnes[i+1];
                ret = max(ret, tmp);
            }
        }
        
        if(ret == 0) {
            return nums[0] == 0 ? 0 : len-1;
        }
        
        return ret;
    }
};

File: No_1493/Longest Subarray of 1's After Deleting One Element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        left = 0
        zero_count = 0
        ans = 0

        for right in range(len(nums)):
            if nums[right] == 0:
                zero_count += 1

            while zero_count > 1:
                if nums[left] == 0:
                    zero_count -= 1
                left += 1

            ans = max(ans, right - left)
        return ans